tests/plugins/imagebase/features/upload.js

```diff
@@ -608,6 +646,56 @@
 						}
 					} );
 				} );
+			},
+
+			'test progressbar does not create snapshots during upload': function() {
+				var editor = this.editor;
+
+				editor.undoManager.reset();
+
+				assertPasteFiles( editor, {
+					files: [ getTestBmpFile() ],
+					fullLoad: false,
+					callback: function( widgets ) {
+						var initialSnapshots = editor.undoManager.snapshots.length,
+							loader = editor.uploadRepository.loaders[ 0 ];
+
+						assert.areSame( 1, widgets.length, 'Widgets count' );
+
+						loader.uploadTotal = 10;
+
+						updateLoader( loader, 2, function() {
```

Comment:
This whole part of tests is very repetitve. I'd extract it to dedicated helper function.